src/Network.cpp

```diff
@@ -981,12 +967,23 @@ void Network::show_heatmap(const FastState* const state,
     for (int i = display_map.size() - 1; i >= 0; --i) {
         myprintf("%s\n", display_map[i].c_str());
     }
-    assert(result.first.back().second == FastBoard::PASS);
-    const auto pass_score = int(result.first.back().first * 1000);
+    const auto pass_score = int(result.policy_pass * 1000);
     myprintf("pass: %d\n", pass_score);
-    myprintf("winrate: %f\n", result.second);
+    myprintf("winrate: %f\n", result.winrate);
 
     if (topmoves) {
+        using ScoreVertexPair = std::pair<float,int>;
```

Comment:
You don't need to redefine this?

Reply:
I'd reuse the typedef (even if the Vertex is not really the same), and put it wherever is most appropriate.